Flaky test_belleii.py::test_dirac_addfile_with_parents_meta

Flaky test runs of test_dirac_addfile_with_parents_meta show the following error:

```
tests/test_belleii.py:118: in test_dirac_addfile_with_parents_meta
    dirac_client.add_files(lfns=lfns, ignore_availability=False, parents_metadata=parents_metadata)
lib/rucio/client/diracclient.py:125: in add_files
    raise exc_cls(exc_msg)
E   rucio.common.exception.InvalidRSEExpression: Provided RSE expression is considered invalid.
E   Details: RSE Expression resulted in an empty set.
```

See also in the logs:

* https://github.com/rucio/rucio/commit/f371145d28347759b84b964f65e171bf958f662d/checks/77393880989/logs
* https://github.com/rucio/rucio/commit/c7cab8a5bb69a579da6aa4104c88bca7ebb4f2e6/checks/77898359439/logs

--- a/lib/rucio/common/cache.py
+++ b/lib/rucio/common/cache.py
@@ -42,6 +42,16 @@ finally:
         _mc_client.close()
 
 
+def flush_memcache() -> bool:
+    import pymemcache
+
+    client = pymemcache.Client(CACHE_URL, connect_timeout=3, timeout=3)
+    try:
+        return client.flush_all()
+    finally:
+        client.close()
+
+
 class MemcacheRegion(CacheRegion):
     """
     Subclass of CacheRegion.

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import functools
+import logging
 import os
 import re
 import traceback
@@ -61,6 +62,7 @@ def pytest_configure(config: pytest.Config) -> None:
         "markers",
         "flaky(reruns, reruns_delay): mark test as flaky and rerun on failure"
     )
+    config.addinivalue_line('markers', 'clear_cache(reason): clears memcache before running the test')
 
     # disable configuration cache globally. Fixes side effects when using, e.g., core_config_mock
     os.environ['RUCIO_CONFIG_DISABLE_CACHE_READ'] = 'true'
@@ -70,10 +72,15 @@ def pytest_configure(config: pytest.Config) -> None:
         config.pluginmanager.register(xdist_noparallel_scheduler)
 
 
+def pytest_collection_modifyitems(items: list[pytest.Item]):
+    for item in items:
+        # automatically mark tests as noparallel, when clear_cache is used
+        if item.get_closest_marker('clear_cache'):
+            item.add_marker(pytest.mark.noparallel(reason='clear_cache marker is applied'))
+
+
 def pytest_runtest_setup(item: pytest.Item) -> None:
-    """
-    Skip tests marked with 'needs_iam' if the IAM profile is not available in the dev environment.
-    """
+    # Skip tests marked with 'needs_iam' if the IAM profile is not available in the dev environment.
     if item.get_closest_marker("needs_iam"):
         # Check if we're running in a dev environment with IAM profile available
         dev_profiles = environ.get('DEV_PROFILES', '').split(',')
@@ -81,6 +88,12 @@ def pytest_runtest_setup(item: pytest.Item) -> None:
 
         if 'iam' not in dev_profiles:
             pytest.skip("Test requires IAM profile - start dev environment with: --profile iam")
+
+    # Flush memcache before test, if clear_cache marker is found
+    if item.get_closest_marker('clear_cache'):
+        from rucio.common.cache import flush_memcache
+        if not flush_memcache():
+            logging.warning("Failed to flush memcache when handling pytest clear_cache marker")
 
 
 def pytest_make_parametrize_id(

--- a/tests/test_belleii.py
+++ b/tests/test_belleii.py
@@ -27,6 +27,7 @@ from rucio.tests.common import did_name_generator, skip_non_belleii
 
 @skip_non_belleii
 @pytest.mark.noparallel(reason='changes global configuration value')
+@pytest.mark.clear_cache(reason='requires clean parse_expression result for ANY=true')
 @pytest.mark.parametrize("file_lifetime", [86400.0, None])
 def test_dirac_addfile(rse_factory, did_factory, root_account, did_client, dirac_client, rse_client, replica_client, file_lifetime):
     """ DIRAC (CLIENT): Test the functionality of the addfile method """
@@ -99,6 +100,8 @@ def test_dirac_addfile(rse_factory, did_factory, root_account, did_client, dirac
 
 
 @skip_non_belleii
+@pytest.mark.noparallel(reason='changes global configuration value')
+@pytest.mark.clear_cache(reason='requires clean parse_expression result for ANY=true')
 def test_dirac_addfile_with_parents_meta(rse_factory, did_factory, root_account, did_client, dirac_client, rse_client, replica_client):
     """ DIRAC (CLIENT): Test the functionality of the addfile method """
     rse1, rse1_id = rse_factory.make_srm_rse(deterministic=True)

